[DR-520] docs: flat release-job validation config reference (#10469)
circleci/circleci-docs: docs/reference/modules/ROOT/pages/configuration-reference.adoc

[#job-name]
=== *<``job_name``>*

> Each job consists of the job's name as a key and a map as a value. A name must be case insensitive unique within a current `jobs` list. The value map has the following attributes:

Reviewer: This sounds a bit awkward, maybe...
```suggestion
Each job consists of the job's name as a key and a map as a value. A name must be unique within a current `jobs` list. Job names are case insensitive. The value map has the following attributes:
```

    plan_name: <my-service-release>
----

Release jobs can optionally enable the deploy validation engine with a flat configuration (no nested `validation_policy` block).

> * `validation.enabled` — when `true` and no `webhooks` or `agents` are listed, creates a default Datadog webhook validation named `default`.

Reviewer: ```suggestion
* `validation.enabled`: When `true` and no `webhooks` or `agents` are listed, creates a default Datadog webhook validation named `default`.
```

> * `webhooks` — optional list of webhook checks. Each entry requires `name` and `type` (`datadog` or `custom`).

Reviewer: ```suggestion
* `webhooks`: Optional list of webhook checks. Each entry requires `name` and `type`. The `type` can be one of either `datadog` or `custom`.
```

> * `agents` — optional list of agentic checks. Each entry requires `name` and `prompt`; `tools` (default `datadog`) and `frequency` (default `5m`) are optional.

Reviewer: ```suggestion
* `agents`: Optional list of agentic checks. Each entry requires `name` and `prompt`; `tools` (default `datadog`) and `frequency` (default `5m`) are optional.
```

Reviewer: Might put these in a table actually

> Validation runs when any of `validation.enabled`, a non-empty `webhooks` list, or a non-empty `agents` list is set.

Reviewer: ```suggestion
Validation runs when any of the following are set: 

* `validation.enabled` is true. 
* A non-empty `webhooks` list. 
* A non-empty `agents` list.
```
